share/sql/share.go

```diff
@@ -345,15 +310,17 @@ func (m *mgr) isProjectAdmin(u *userpb.User, path string) bool {
 	return false
 }
 
-func (m *mgr) ListShares(ctx context.Context, filters []*collaboration.Filter) ([]*collaboration.Share, error) {
-	uid := conversions.FormatUserID(appctx.ContextMustGetUser(ctx).Id)
-
+func (m *shareMgr) ListShares(ctx context.Context, filters []*collaboration.Filter) ([]*collaboration.Share, error) {
 	query := m.db.Model(&model.Share{}).
-		Where("uid_owner = ? or uid_initiator = ?", uid, uid).
 		Where("orphan = ?", false)
 
+	if u, ok := appctx.ContextGetUser(ctx); ok {
+		uid := conversions.FormatUserID(u.Id)
+		query = query.Where("uid_owner = ? or uid_initiator = ?", uid, uid)
```

Comment:
For consistency, you should use `Filter_TYPE_OWNER` and (I suppose) `Filter_TYPE_CREATOR` and add those options in `appendShareFiltersToQuery`.

Reply:
Ah, although here you need a "or"......

Reply:
But maybe it's better to update `ListPublicShares` then to also use `uid_owner = ? or uid_initiator = ?`?

Reply:
Yes, please update `ListPublicShares` to make it consistent.